src/orion/benchmark/task/quadratics.py

```diff
@@ -0,0 +1,505 @@
+            seed=self.seed,
+            max_trials=self.max_trials,
+        )
+        # Note: rounding to 4 decimals just to prevent some potential issues
+        # related to the string representation of the task later on.
+        self.a2 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a2 is None else self.a2, 4
+        )
+        self.a1 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a1 is None else self.a1, 4
+        )
+        self.a0 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a0 is None else self.a0, 4
+        )
+
+    def with_context(self) -> "QuadraticsTaskWithContext":
+        return QuadraticsTaskWithContext(
+            a2=self.a2, a1=self.a1, a0=self.a0, task_id=self.task_id, rng=self.rng,
+        )
+
+    def __repr__(self) -> str:
+        return f"QuadraticsTask(a0={self.a0}, a1={self.a1}, a2={self.a2})"
+
+    def __call__(
+        self,
+        hp: Union[
+            QuadraticsTaskHParams, List[QuadraticsTaskHParams], np.ndarray
+        ] = None,
+        **kwargs,
+    ) -> np.ndarray:
+        if hp is None and kwargs:
+            hp = self.hparams(**kwargs)
+        if (isinstance(hp, list) and hp and not isinstance(hp[0], float)) or (
+            isinstance(hp, np.ndarray) and hp.ndim == 2
+        ):
+            # This is being called with a batched input.
+            return np.array(list(map(self, hp)))
+        x: np.ndarray
+        if isinstance(hp, np.ndarray):
+            x = hp
+        elif isinstance(hp, QuadraticsTaskHParams):
+            x = hp.to_array()
+        else:
+            x = np.asfarray(hp)
+
+        if x.shape[-1] > 3:
+            # Discard any 'extra' values.
+            # NOTE: This is only really used in the 'with context' version of
+            # this task below.
+            x = x[..., :3]
+
+        assert x.shape == (3,)
+        y = 0.5 * self.a2 * (x ** 2).sum() + self.a1 * x.sum() + self.a0
+
+        # Interesting: This polynomial has the same result, when passed x.sum(-1) rather
+        # than `x`.
+        # from numpy.polynomial import Polynomial
+        # poly = Polynomial(
+        #     coef=[self.a0, self.a1, self.a2],
+        #     domain=[0.1, 10.0],
+        #     window=[0.1, 10.0],
+        # )
+        # y_poly = poly(x.sum(-1))
+        # assert False, (y, y_poly)
+
+        return y
+
+    def gradient(self, x: np.ndarray) -> float:
+        # if x.ndim == 2:
+        #     assert x.shape[-1] == 3
+        #     return self.a2 * x.sum(-1) + self.a1
+        assert x.shape[-1] == 3
+        return self.a2 * x.sum(-1) + self.a1
+
+    @property
+    def hash(self) -> str:
+        # TODO: Return a unique "hash"/id/key for this task
+
+        if is_dataclass(self):
+            return compute_identity(**asdict(self))
+        return compute_identity(
+            **self.space, a0=self.a0, a1=self.a1, a2=self.a2, task_id=self.task_id
+        )
+
+    def __neg__(self) -> "QuadraticsTask":
```

Comment:
Where are these operations (neg, add, sub) used?

Reply:
They were being used when trying to sample a task with a given similarity or distance from another task. 
I removed all of that code, so I also removed all of these.